apps/prairielearn/src/lib/course-request.ts

```diff
@@ -0,0 +1,104 @@
+import { loadSqlEquiv, queryRow, queryAsync } from '@prairielearn/postgres';
+import { z } from 'zod';
+import { logger } from '@prairielearn/logger';
+import * as Sentry from '@prairielearn/sentry';
+
+import { DateFromISOString, IdSchema } from '../lib/db-types';
+import { createCourseRepoJob } from '../lib/github';
+import { sendCourseRequestMessage } from '../lib/opsbot';
+
+const sql = loadSqlEquiv(__filename);
+
+const JobsRowSchema = z.object({
+  authn_user_id: IdSchema.optional(),
+  finish_date: DateFromISOString.optional(),
+  id: IdSchema.optional(),
+  number: z.number().optional(),
+  start_date: DateFromISOString.optional(),
+  status: z.string().optional(),
```

Comment:
I imagine you marked these as optional because `jobs` was declared as a (sometimes-empty) object instead of an array. Once you change `jobs` to `z.array(JobsRowSchema)`, you should revisit this schema to determined which of these, if any, are actually optional/nullable.

Reply:
Yes, that is exactly the reasoning. I have now updated jobs to `z.array(JobsRowSchema)`. I added `.optional()` to the JobsRowSchema object itself and removed them these values inside the schema. I did add `.nullable()` to authn_user_id, authn_user_name, and finish_date as it appears these fields could possibly contain null values.